DataFormats/TrackerRecHit2D/interface/SiPixelRecHitQuality.h

```diff
@@ -123,6 +123,9 @@ class SiPixelRecHitQuality {
     inline void setProbabilityQ(float prob, QualWordType& qualWord) const {
       if (prob > 1.0f && prob <= 1.0f + std::numeric_limits<float>::epsilon()) {
         prob = 1;
+      } else if (prob == -1.0f) {
```

Comment:
Can we make this a more numerically safe comparison? 

```
std::abs( prob + 1.0f) < std::numeric_limits<float>::epsilon()
```

Reply:
Here is it meant to refer to only one specific case of the default value. So, there is no way to get a different value. 
Addition of a rounding condition here would cover unintentionally some other possible calls to `setProbabilityQ`.
Note that the 1+epsilon in the part above was intentional due to function calls leading to rounding errors in the prob **computation**.

**HOWEVER**, while investigating this point, I see that there is a ```hasFilledProb_``` in the CPE computation that controlls if the setProb is going to be called or not.
So, we perhaps should revert a change in `setProbabilityQ` and propagate a correct setting of ```hasFilledProb_```  in the CPE implementation(s).

Reply:
This is now done in [ac5fe99](https://github.com/cms-sw/cmssw/pull/39776/commits/ac5fe997b70b12757f07661bef39deb14352daeb).

Reply:
After a further inspection, I'm reverting back to the original comment ; `hasFilledProb_` controls way more than what we wanted to address with `setProbabilityQ`.
We should just go back to https://github.com/cms-sw/cmssw/commit/bd6dab721918035528b1471ff9db5760384e1d52

Sorry for the random digression.

Reply:
The reset to (bd6dab7)[https://github.com/cms-sw/cmssw/commit/bd6dab721918035528b1471ff9db5760384e1d52] is done.